### PR TITLE
Abstraction and on disk implementation of blob storage

### PR DESCRIPTION
**Pull request by shubham3121**
## Description
Please include a summary of the change, the motivation, and any additional context that will help others understand your PR. If it closes one or more open issues, [please tag them as described here](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

## Affected Dependencies
List any dependencies that are required for this change.

## How has this been tested?
- Describe the tests that you ran to verify your changes.
- Provide instructions so we can reproduce.
- List any relevant details for your test configuration.

## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [x] My changes are covered by tests

### CODE REVIEW

**Comment by rasswanth-s** [comment]
Blob Storage 🙌 🙌 🙌  !!!

**Comment by kiendang** [comment]
@shubham3121 @PeterChung241 @khoaguin I've finished renaming all the file stuff to blob storage as we discussed. Please review all the commits starting with `Rename ...`.

There are 2 changes I made that are different from what we discussed. One is renaming `SyftResource`  and `SyftWriteResource` to `BlobRetrieval` and `BlobDeposit` instead of `SyftReadResource`  and `SyftWriteResource`. Alr discussed with Shubham about this.

The second one is renaming `FileObject` to `BlobStorageEntry` instead of `BlobStorageInformation`. Reasons are:
- We have things like `CreateFileObject`, `get_file_object_by_uid`, `get_all_file_object`, ... Replacing `file_object` here with `blob_storage_entry` instead of `blob_storage_information` reads better grammatically
- The object works like an entry in the blob storage as well since it serves as a record to help allocate and keep track of the syft object in the blob storage.

Let me know what you think. Happy to make changes again.

**Comment by kiendang** [comment]
Sorry for abusing force pushing, just to make the commits clean for easier review. If you want to synchronize your local repo with this PR do `git pull --rebase`.

File: packages/syft/src/syft/store/blob_storage/__init__.py [hunk 1]
```diff
@@ -0,0 +1,135 @@
+"""Blob file storage
+
+Contains blob file storage interfaces. See `on_disk.py` for an example of a concrete implementation.
```
Comment (shubham3121) [inline packages/syft/src/syft/store/blob_storage/__init__.py:3]: ❤️

**Comment by shubham3121** [comment]
Excellent work @kiendang !! 💯 🔥 ❤️

**Review by kiendang — APPROVED** [review]
@shubham3121 I realized you created this PR so probably can't approve it? I'm gonna just approve it here. When everything looks good you can merge it.

**Pull request merged**